File: gc-modules/gc-module-system/src/main/java/com/gc/system/service/SysRoleService.java
package com.gc.system.service;

import com.gc.starter.crud.service.BaseService;
import com.gc.system.model.SysRolePO;
import com.gc.system.pojo.dto.role.RoleMenuSaveDTO;

/**
 * @author jackson
 * 2020/1/24 2:20 下午
 */
public interface SysRoleService extends BaseService<SysRolePO> {

    /**
     * 保存角色的菜单信息
     * @param parameter 参数
     * @return 是否保存成功
     */
    boolean saveRoleMenu(RoleMenuSaveDTO parameter);
}

File: smart-module-starter/smart-system/src/main/kotlin/com/smart/system/service/SysRoleService.kt
package com.smart.system.service

import com.smart.auth.common.model.SysUserDO
import com.smart.common.model.Tree
import com.smart.starter.crud.service.BaseService
import com.smart.system.model.SysRoleDO
import com.smart.system.model.SysRoleMenuFunctionDO

/**
 *
 * @author ming
 * 2019/6/12 下午3:48
 */
interface SysRoleService : BaseService<SysRoleDO> {

    fun queryRoleUser(roleIdList: List<String>): Map<String, List<SysUserDO>>
    abstract fun listAllTreeWithOrgan(): Tree<Any>?
    abstract fun updateUser(roleToUserIdList: Map<String, List<String>>): Boolean
    abstract fun authorize(list: List<SysRoleMenuFunctionDO>): Boolean
}